### DISCUSSION TITLE
How do I get local and remote IP addresses?

### DISCUSSION THREAD

**Discussion by lostmsu** [sipsorcery-org/sipsorcery#1013] [Q&A]
After the connection is established, how can I get local and remote IP addresses that were used?

**Comment by imtadashi** [comment]
What do you mean by local and remote ip? Does it mean your ip(local) and other peer ip(remote)?

**Reply by lostmsu** [reply]
Yes, exactly.

**Comment by imtadashi** [comment]
You can create you own program to get your ip and remote ip. To get public ip address you can use public STUN server like stun:stun.l.google.com:19302 or you can host your own using coturn. On connection establishment, you can run a piece of code on the local user side to get his ip and on remote side, you can get their ip and send it via data channel. This is one of the basic way, but there can be more simpler way than this.

**Reply by lostmsu** [reply]
This is not what I mean. I don't need "my/remote ip address pair", I need "my/remote ip address pair that was used to establish this particular instance of `RTCPeerConnection`". E.g. whatever pair of ICE candidates that ended up being used.
